Crash with BUKKIT caused by NPE

https://github.com/Railcraft/Railcraft/blob/mc-1.12.2/src/main/java/mods/railcraft/common/util/misc/Game.java#L69
here invokes `Game.log()` which will invoke `Game.log("default")`
https://github.com/Railcraft/Railcraft/blob/mc-1.12.2/src/main/java/mods/railcraft/common/util/misc/Game.java#L125
and here invokes `enabledCategories::contains` while `enabledCategories` is not initialized yet, and will throw NullPointerException.

I've done an experiment to proof that:

![image](https://user-images.githubusercontent.com/25274081/53890504-4da35300-4063-11e9-8171-9b4d6b6a9013.png)

![image](https://user-images.githubusercontent.com/25274081/53890523-53993400-4063-11e9-9d48-9facf8ed804c.png)

And in order to fix this, you just need to move the declaration before the static code block, or just simply initialize in the static code block.

Game Stack Trace: 
```
net.minecraftforge.fml.common.LoaderExceptionModCrash: Caught exception from Railcraft (railcraft)
Caused by: java.lang.ExceptionInInitializerError
	at mods.railcraft.common.modules.RailcraftModuleManager.loadModules(RailcraftModuleManager.java:55)
	at mods.railcraft.common.core.Railcraft.preInit(Railcraft.java:106)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at net.minecraftforge.fml.common.FMLModContainer.handleModStateEvent(FMLModContainer.java:623)
	at sun.reflect.GeneratedMethodAccessor4.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Subscriber.java:91)
	at com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Subscriber.java:150)
	at com.google.common.eventbus.Subscriber$1.run(Subscriber.java:76)
	at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:399)
	at com.google.common.eventbus.Subscriber.dispatchEvent(Subscriber.java:71)
	at com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Dispatcher.java:116)
	at com.google.common.eventbus.EventBus.post(EventBus.java:217)
	at net.minecraftforge.fml.common.LoadController.sendEventToModContainer(LoadController.java:219)
	at net.minecraftforge.fml.common.LoadController.propogateStateMessage(LoadController.java:197)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Subscriber.java:91)
	at com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Subscriber.java:150)
	at com.google.common.eventbus.Subscriber$1.run(Subscriber.java:76)
	at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:399)
	at com.google.common.eventbus.Subscriber.dispatchEvent(Subscriber.java:71)
	at com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Dispatcher.java:116)
	at com.google.common.eventbus.EventBus.post(EventBus.java:217)
	at net.minecraftforge.fml.common.LoadController.distributeStateMessage(LoadController.java:136)
	at net.minecraftforge.fml.common.Loader.preinitializeMods(Loader.java:627)
	at net.minecraftforge.fml.server.FMLServerHandler.beginServerLoading(FMLServerHandler.java:98)
	at net.minecraftforge.fml.common.FMLCommonHandler.onServerStart(FMLCommonHandler.java:333)
	at net.minecraft.server.dedicated.DedicatedServer.func_71197_b(DedicatedServer.java:166)
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:599)
	at java.lang.Thread.run(Unknown Source)
Caused by: java.lang.NullPointerException
	at mods.railcraft.common.util.misc.Game.log(Game.java:125)
	at mods.railcraft.common.util.misc.Game.log(Game.java:121)
	at mods.railcraft.common.util.misc.Game.<clinit>(Game.java:69)
	... 37 more
```

--- a/build.gradle
+++ b/build.gradle
@@ -70,7 +70,8 @@ def actualRunDir = project.file(minecraft.runDir)
 def optimizedVmArgs = ['-XX:+UseG1GC', '-Xmx4G', '-Xms4G', '-XX:+UnlockExperimentalVMOptions', '-XX:G1NewSizePercent=20', '-XX:G1ReservePercent=20', '-XX:MaxGCPauseMillis=50', '-XX:G1HeapRegionSize=32M']
 def assertionVmArgs = ['-ea:mods.railcraft...', '-ea:net.minecraft...', '-ea:net.minecraftforge...']
 def log4jVmArgs = ['-Dlog4j.configurationFile=' + new File(actualRunDir, 'log4j2.xml').toString()]
-def combinedVmArgs = [assertionVmArgs, optimizedVmArgs, log4jVmArgs].flatten()
+def railcraftDebugArgs = ['-Drailcraft.log.categories=registry models']
+def combinedVmArgs = [assertionVmArgs, optimizedVmArgs, log4jVmArgs, railcraftDebugArgs].flatten()
 def extraEnvs = new LinkedHashMap() //['LOG4J_CONFIGURATION_FILE': "log4j2.xml"]
 
 minecraft {

--- a/build.gradle
+++ b/build.gradle
@@ -70,7 +70,7 @@ def actualRunDir = project.file(minecraft.runDir)
 def optimizedVmArgs = ['-XX:+UseG1GC', '-Xmx4G', '-Xms4G', '-XX:+UnlockExperimentalVMOptions', '-XX:G1NewSizePercent=20', '-XX:G1ReservePercent=20', '-XX:MaxGCPauseMillis=50', '-XX:G1HeapRegionSize=32M']
 def assertionVmArgs = ['-ea:mods.railcraft...', '-ea:net.minecraft...', '-ea:net.minecraftforge...']
 def log4jVmArgs = ['-Dlog4j.configurationFile=' + new File(actualRunDir, 'log4j2.xml').toString()]
-def railcraftDebugArgs = ['-Drailcraft.log.categories=registry models']
+def railcraftDebugArgs = ['-Drailcraft.log.categories=registry,models']
 def combinedVmArgs = [assertionVmArgs, optimizedVmArgs, log4jVmArgs, railcraftDebugArgs].flatten()
 def extraEnvs = new LinkedHashMap() //['LOG4J_CONFIGURATION_FILE': "log4j2.xml"]
 

--- a/src/main/java/mods/railcraft/common/carts/CartTools.java
+++ b/src/main/java/mods/railcraft/common/carts/CartTools.java
@@ -29,6 +29,7 @@ import net.minecraft.entity.player.EntityPlayerMP;
 import net.minecraft.item.Item;
 import net.minecraft.item.ItemStack;
 import net.minecraft.util.EnumFacing;
+import net.minecraft.util.EnumHand;
 import net.minecraft.util.math.AxisAlignedBB;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.math.Vec3d;
@@ -156,7 +157,17 @@ public final class CartTools {
         if (!RailcraftConstantsAPI.UNKNOWN_PLAYER.equals(owner.getName()))
             player = PlayerPlugin.getPlayer(cart.world, owner);
         if (player == null)
-            player = RailcraftFakePlayer.get((WorldServer) cart.world, cart.posX, cart.posY, cart.posZ);
+            player = getFakePlayer(cart);
+        return player;
+    }
+
+    public static EntityPlayerMP getFakePlayer(EntityMinecart cart) {
+        return RailcraftFakePlayer.get((WorldServer) cart.world, cart.posX, cart.posY, cart.posZ);
+    }
+
+    public static EntityPlayerMP getFakePlayerWith(EntityMinecart cart, ItemStack stack) {
+        EntityPlayerMP player = getFakePlayer(cart);
+        player.setHeldItem(EnumHand.MAIN_HAND, stack);
         return player;
     }
 

--- a/src/main/java/mods/railcraft/common/carts/EntityTunnelBore.java
+++ b/src/main/java/mods/railcraft/common/carts/EntityTunnelBore.java
@@ -823,7 +823,7 @@ public class EntityTunnelBore extends CartBaseContainer implements ILinkableCart
         EntityPlayerMP fakePlayer = CartTools.getFakePlayerWith(this, head);
 
         // Fires break event within; harvest handled separately
-        BlockEvent.BreakEvent breakEvent = new BlockEvent.BreakEvent(world, targetPos, targetState, CartTools.getCartOwnerEntity(this));
+        BlockEvent.BreakEvent breakEvent = new BlockEvent.BreakEvent(world, targetPos, targetState, fakePlayer);
         MinecraftForge.EVENT_BUS.post(breakEvent);
 
         if (breakEvent.isCanceled())

--- a/src/main/java/mods/railcraft/common/carts/EntityTunnelBore.java
+++ b/src/main/java/mods/railcraft/common/carts/EntityTunnelBore.java
@@ -41,6 +41,7 @@ import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.entity.item.EntityItem;
 import net.minecraft.entity.item.EntityMinecart;
 import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.entity.player.EntityPlayerMP;
 import net.minecraft.init.Blocks;
 import net.minecraft.init.Enchantments;
 import net.minecraft.init.Items;
@@ -49,10 +50,7 @@ import net.minecraft.item.ItemStack;
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.network.datasync.DataParameter;
 import net.minecraft.network.datasync.DataSerializers;
-import net.minecraft.util.DamageSource;
-import net.minecraft.util.EnumFacing;
-import net.minecraft.util.EnumParticleTypes;
-import net.minecraft.util.NonNullList;
+import net.minecraft.util.*;
 import net.minecraft.util.math.AxisAlignedBB;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.math.MathHelper;
@@ -513,7 +511,7 @@ public class EntityTunnelBore extends CartBaseContainer implements ILinkableCart
 
                 ItemStack head = getStackInSlot(0);
                 if (!InvTools.isEmpty(head)) {
-                    head.damageItem(entities.size(), CartTools.getCartOwnerEntity(this));
+                    head.damageItem(entities.size(), CartTools.getFakePlayer(this));
                 }
             }
 
@@ -698,7 +696,7 @@ public class EntityTunnelBore extends CartBaseContainer implements ILinkableCart
                             IBlockState state = WorldPlugin.getBlockState(world, searchPos);
                             if (!WorldPlugin.isBlockAir(world, searchPos, state) && !state.getMaterial().isLiquid()) {
                                 // Break other blocks first
-                                WorldPlugin.playerRemoveBlock(world, searchPos.toImmutable(), CartTools.getCartOwnerEntity(this),
+                                WorldPlugin.playerRemoveBlock(world, searchPos.toImmutable(), CartTools.getFakePlayer(this),
                                         world.getGameRules().getBoolean("doTileDrops") && RailcraftConfig.borePreserveStacks());
                             }
                         }
@@ -717,7 +715,7 @@ public class EntityTunnelBore extends CartBaseContainer implements ILinkableCart
     }
 
     protected boolean placeTrack(BlockPos targetPos, IBlockState oldState, BlockRailBase.EnumRailDirection shape) {
-        EntityPlayer owner = CartTools.getCartOwnerEntity(this);
+        EntityPlayer owner = CartTools.getFakePlayer(this);
 
         if (replaceableBlocks.contains(oldState.getBlock()))
             WorldPlugin.destroyBlockSafe(world, targetPos, owner, true);
@@ -820,8 +818,10 @@ public class EntityTunnelBore extends CartBaseContainer implements ILinkableCart
         if (!canMineBlock(targetPos, targetState))
             return false;
 
+        EntityPlayerMP fakePlayer = CartTools.getFakePlayerWith(this, head);
+
         // Start of Event Fire
-        BreakEvent breakEvent = new BreakEvent(world, targetPos, targetState, CartTools.getCartOwnerEntity(this));
+        BreakEvent breakEvent = new BreakEvent(world, targetPos, targetState, fakePlayer);
         MinecraftForge.EVENT_BUS.post(breakEvent);
 
         if (breakEvent.isCanceled())
@@ -843,7 +843,7 @@ public class EntityTunnelBore extends CartBaseContainer implements ILinkableCart
         }
 
         // Start of Event Fire
-        BlockEvent.HarvestDropsEvent harvestDropsEvent = new BlockEvent.HarvestDropsEvent(world, targetPos, targetState, fortuneLevel, 1F, items, CartTools.getCartOwnerEntity(this), silk);
+        BlockEvent.HarvestDropsEvent harvestDropsEvent = new BlockEvent.HarvestDropsEvent(world, targetPos, targetState, fortuneLevel, 1F, items, fakePlayer, silk);
         MinecraftForge.EVENT_BUS.post(harvestDropsEvent);
 
         if (harvestDropsEvent.isCanceled())
@@ -875,7 +875,7 @@ public class EntityTunnelBore extends CartBaseContainer implements ILinkableCart
         }
         WorldPlugin.setBlockToAir(world, targetPos);
 
-        head.damageItem(1, CartTools.getCartOwnerEntity(this));
+        head.damageItem(1, fakePlayer);
         if (head.getItemDamage() > head.getMaxDamage())
             setInventorySlotContents(0, ItemStack.EMPTY);
         return true;

--- a/src/main/java/mods/railcraft/common/carts/EntityTunnelBore.java
+++ b/src/main/java/mods/railcraft/common/carts/EntityTunnelBore.java
@@ -50,7 +50,10 @@ import net.minecraft.item.ItemStack;
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.network.datasync.DataParameter;
 import net.minecraft.network.datasync.DataSerializers;
-import net.minecraft.util.*;
+import net.minecraft.util.DamageSource;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.EnumParticleTypes;
+import net.minecraft.util.NonNullList;
 import net.minecraft.util.math.AxisAlignedBB;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.math.MathHelper;
@@ -59,7 +62,6 @@ import net.minecraft.world.World;
 import net.minecraft.world.WorldServer;
 import net.minecraftforge.common.MinecraftForge;
 import net.minecraftforge.event.world.BlockEvent;
-import net.minecraftforge.event.world.BlockEvent.BreakEvent;
 import org.jetbrains.annotations.Contract;
 import org.jetbrains.annotations.Nullable;
 
@@ -820,37 +822,32 @@ public class EntityTunnelBore extends CartBaseContainer implements ILinkableCart
 
         EntityPlayerMP fakePlayer = CartTools.getFakePlayerWith(this, head);
 
-        // Start of Event Fire
-        BreakEvent breakEvent = new BreakEvent(world, targetPos, targetState, fakePlayer);
-        MinecraftForge.EVENT_BUS.post(breakEvent);
-
-        if (breakEvent.isCanceled())
+        // Fires break event within; harvest handled separately
+        if (!WorldPlugin.playerRemoveBlock(world, targetPos, fakePlayer, false))
             return false;
-        // End of Event Fire
-
-        boolean silk = EnchantmentHelper.getEnchantmentLevel(Enchantments.SILK_TOUCH, head) > 0;
-        NonNullList<ItemStack> items = NonNullList.create();
-        int fortuneLevel = EnchantmentHelper.getEnchantmentLevel(Enchantments.FORTUNE, head);
-
-        if (silk) {
-            ItemStack stack = HarvestPlugin.getSilkTouchDrop(targetState);
-            if (!InvTools.isEmpty(stack)) {
-                items.add(stack);
-            }
-            // Use modifiable lists for events
-        } else {
-            targetState.getBlock().getDrops(items, world, targetPos, targetState, fortuneLevel);
-        }
-
-        // Start of Event Fire
-        BlockEvent.HarvestDropsEvent harvestDropsEvent = new BlockEvent.HarvestDropsEvent(world, targetPos, targetState, fortuneLevel, 1F, items, fakePlayer, silk);
-        MinecraftForge.EVENT_BUS.post(harvestDropsEvent);
-
-        if (harvestDropsEvent.isCanceled())
-            return false;
-        // End of Event Fire
 
         if (RailcraftConfig.borePreserveStacks() && world.getGameRules().getBoolean("doTileDrops")) {
+            boolean silk = EnchantmentHelper.getEnchantmentLevel(Enchantments.SILK_TOUCH, head) > 0;
+            NonNullList<ItemStack> items = NonNullList.create();
+            int fortuneLevel = EnchantmentHelper.getEnchantmentLevel(Enchantments.FORTUNE, head);
+
+            if (silk) {
+                ItemStack stack = HarvestPlugin.getSilkTouchDrop(targetState);
+                if (!InvTools.isEmpty(stack)) {
+                    items.add(stack);
+                }
+                // Use modifiable lists for events
+            } else {
+                targetState.getBlock().getDrops(items, world, targetPos, targetState, fortuneLevel);
+            }
+            // Start of Event Fire
+            BlockEvent.HarvestDropsEvent harvestDropsEvent = new BlockEvent.HarvestDropsEvent(world, targetPos, targetState, fortuneLevel, 1F, items, fakePlayer, silk);
+            MinecraftForge.EVENT_BUS.post(harvestDropsEvent);
+
+            if (harvestDropsEvent.isCanceled())
+                return false;
+            // End of Event Fire
+
             for (ItemStack stack : items) {
                 if (StackFilters.FUEL.test(stack))
                     stack = invFuel.addStack(stack);
@@ -873,7 +870,6 @@ public class EntityTunnelBore extends CartBaseContainer implements ILinkableCart
                 }
             }
         }
-        WorldPlugin.setBlockToAir(world, targetPos);
 
         head.damageItem(1, fakePlayer);
         if (head.getItemDamage() > head.getMaxDamage())

--- a/src/main/java/mods/railcraft/common/carts/EntityTunnelBore.java
+++ b/src/main/java/mods/railcraft/common/carts/EntityTunnelBore.java
@@ -823,7 +823,10 @@ public class EntityTunnelBore extends CartBaseContainer implements ILinkableCart
         EntityPlayerMP fakePlayer = CartTools.getFakePlayerWith(this, head);
 
         // Fires break event within; harvest handled separately
-        if (!WorldPlugin.playerRemoveBlock(world, targetPos, fakePlayer, false))
+        BlockEvent.BreakEvent breakEvent = new BlockEvent.BreakEvent(world, targetPos, targetState, CartTools.getCartOwnerEntity(this));
+        MinecraftForge.EVENT_BUS.post(breakEvent);
+
+        if (breakEvent.isCanceled())
             return false;
 
         if (RailcraftConfig.borePreserveStacks() && world.getGameRules().getBoolean("doTileDrops")) {
@@ -870,6 +873,8 @@ public class EntityTunnelBore extends CartBaseContainer implements ILinkableCart
                 }
             }
         }
+
+        WorldPlugin.setBlockToAir(world, targetPos);
 
         head.damageItem(1, fakePlayer);
         if (head.getItemDamage() > head.getMaxDamage())

--- a/src/main/java/mods/railcraft/common/carts/ItemBoreHead.java
+++ b/src/main/java/mods/railcraft/common/carts/ItemBoreHead.java
@@ -25,6 +25,7 @@ import net.minecraft.item.ItemStack;
 import net.minecraft.item.ItemTool;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.world.World;
+import net.minecraftforge.common.util.FakePlayer;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 import org.jetbrains.annotations.Nullable;
@@ -60,7 +61,9 @@ public abstract class ItemBoreHead extends ItemTool implements IBoreHead, IRailc
 
     @Override
     public int getHarvestLevel(ItemStack stack, String toolClass, @Nullable EntityPlayer player, @Nullable IBlockState blockState) {
-        return IBoreHead.super.getHarvestLevel(stack, toolClass, player, blockState);
+        if (player instanceof FakePlayer)
+            return IBoreHead.super.getHarvestLevel(stack, toolClass, player, blockState);
+        return -1;
     }
 
     @Override

--- a/src/main/java/mods/railcraft/common/util/charge/ChargeManager.java
+++ b/src/main/java/mods/railcraft/common/util/charge/ChargeManager.java
@@ -47,7 +47,6 @@ public enum ChargeManager implements Charge.IManager {
 
     @Override
     public Charge.INetwork network(World world) {
-        Game.notClient(world);
-        return networks.computeIfAbsent(world, (w) -> new ChargeNetwork(type, w));
+        return networks.computeIfAbsent(Game.requireHost(world), (w) -> new ChargeNetwork(type, w));
     }
 }

--- a/src/main/java/mods/railcraft/common/util/misc/Game.java
+++ b/src/main/java/mods/railcraft/common/util/misc/Game.java
@@ -9,13 +9,15 @@
  -----------------------------------------------------------------------------*/
 package mods.railcraft.common.util.misc;
 
+import com.google.common.collect.Sets;
 import mods.railcraft.api.core.ClientAccessException;
 import mods.railcraft.common.core.Railcraft;
 import net.minecraft.client.multiplayer.WorldClient;
-import net.minecraft.entity.Entity;
+import net.minecraft.launchwrapper.Launch;
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.util.ResourceLocation;
 import net.minecraft.world.World;
+import net.minecraft.world.WorldServer;
 import net.minecraftforge.fml.client.FMLClientHandler;
 import net.minecraftforge.fml.common.FMLCommonHandler;
 import net.minecraftforge.fml.common.Loader;
@@ -29,8 +31,7 @@ import org.apache.logging.log4j.message.MessageFormatMessage;
 import org.apache.logging.log4j.message.SimpleMessage;
 import org.jetbrains.annotations.Nullable;
 
-import java.util.Arrays;
-import java.util.HashSet;
+import java.util.Collections;
 import java.util.Set;
 
 /**
@@ -45,19 +46,20 @@ public final class Game {
      */
     public static final Level DEBUG_REPORT = Level.forName("DEBUG_REPORT", 150);
 
+    /**
+     * Separate options with spaces.
+     */
+    private static final String CATEGORY_SETTINGS = "railcraft.log.categories";
+    /**
+     * Other options include "registry" and "models" for now.
+     */
+    private static final Set<String> enabledCategories = Sets.newHashSet("default");
+
+    private static ILogger NULL_LOGGER = new ILogger() {};
+
     static {
-        boolean worldFound = false;
-        boolean worldObjFound = false;
-        try {
-            worldFound = Entity.class.getDeclaredField("world") != null;
-        } catch (NoSuchFieldException | SecurityException ignored) {
-        }
-        try {
-            //noinspection JavaReflectionMemberAccess
-            worldObjFound = Entity.class.getDeclaredField("worldObj") != null;
-        } catch (NoSuchFieldException | SecurityException ignored) {
-        }
-        OBFUSCATED = !worldFound && !worldObjFound;
+        Object obj = Launch.blackboard.get("fml.deobfuscatedEnvironment");
+        OBFUSCATED = obj instanceof Boolean && ((Boolean) obj);
         DEVELOPMENT_VERSION = Railcraft.getVersion().matches(".*(alpha|beta).*") || !OBFUSCATED;
         boolean foundBukkit = false;
         try {
@@ -67,6 +69,11 @@ public final class Game {
         BUKKIT = foundBukkit;
         if (BUKKIT)
             Game.log().msg(Level.INFO, "Bukkit detected, disabling Tile Entity caching because Bukkit doesn't seem to invalid Tile Entities properly!");
+
+        String st = System.getProperty(CATEGORY_SETTINGS);
+        if (st != null) {
+            Collections.addAll(enabledCategories, st.split(" "));
+        }
     }
 
     public static boolean isHost(final World world) {
@@ -77,8 +84,9 @@ public final class Game {
         return world.isRemote;
     }
 
-    public static void notClient(final World world) {
+    public static WorldServer requireHost(final World world) {
         if (isClient(world)) throw new ClientAccessException();
+        return (WorldServer) world;
     }
 
     public static void requiresServerThread() {
@@ -110,13 +118,6 @@ public final class Game {
     private Game() {
     }
 
-    @SuppressWarnings("ArraysAsListWithZeroOrOneArgument")
-    private static final Set<String> enabledCategories = new HashSet<>(Arrays.asList(
-//                "registry",
-//                "models",
-            "default"
-    ));
-
     public static ILogger log() {
         return log("default");
     }
@@ -126,8 +127,6 @@ public final class Game {
             return Logger.INSTANCE;
         return NULL_LOGGER;
     }
-
-    private static ILogger NULL_LOGGER = new ILogger() {};
 
     public enum Logger implements ILogger {
         INSTANCE;

--- a/src/main/java/mods/railcraft/common/util/misc/Game.java
+++ b/src/main/java/mods/railcraft/common/util/misc/Game.java
@@ -47,7 +47,7 @@ public final class Game {
     public static final Level DEBUG_REPORT = Level.forName("DEBUG_REPORT", 150);
 
     /**
-     * Separate options with spaces.
+     * Separate options with commas.
      */
     private static final String CATEGORY_SETTINGS = "railcraft.log.categories";
     /**
@@ -72,7 +72,7 @@ public final class Game {
 
         String st = System.getProperty(CATEGORY_SETTINGS);
         if (st != null) {
-            Collections.addAll(enabledCategories, st.split(" "));
+            Collections.addAll(enabledCategories, st.split(","));
         }
     }
 

--- a/src/main/java/mods/railcraft/common/util/misc/Game.java
+++ b/src/main/java/mods/railcraft/common/util/misc/Game.java
@@ -59,7 +59,7 @@ public final class Game {
 
     static {
         Object obj = Launch.blackboard.get("fml.deobfuscatedEnvironment");
-        OBFUSCATED = obj instanceof Boolean && ((Boolean) obj);
+        OBFUSCATED = !(obj instanceof Boolean && ((Boolean) obj));
         DEVELOPMENT_VERSION = Railcraft.getVersion().matches(".*(alpha|beta).*") || !OBFUSCATED;
         boolean foundBukkit = false;
         try {

